src/chase/impl/reference.rs

```diff
@@ -6,19 +6,47 @@ use crate::chase::{
     , E, Rel, Observation, WitnessTermTrait, ModelTrait
     , SelectorTrait, EvaluatorTrait, BounderTrait};
 use std::fmt;
-use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::collections::{HashMap, HashSet};
 use itertools::{Itertools, Either};
 use std::iter::FromIterator;
+use std::hash::{Hash, Hasher};
 
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Element(Rc<Cell<E>>);
```

Comment:
Consider implementing `Deref<Target=Rc<Cell<E>>>` for this type, so that you don't need to define `get` nor `replace`.

Reply:
Good idea! will do